src/components/modalWindows/addMovie/addMovie.js

```diff
@@ -1,151 +1,193 @@
 import React, { Fragment, useCallback } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+
+import { addMovie, updateMovie } from './../../../services/requests';
+
+import ErrorMessage from '../../errorMessage';
 
 import './addMovie.css';
 
   const AddMovie = (props) => {
 
     const {
-      id,
-      title,
-      data,
-      genre,
-      url,
-      overview,
-      runtime,
+      activeId,
+      movie,
       handleClose,
-      editMovie
+      editMovie,
+      addNewMovie, 
+      editExistMovie,
     } = props;
-  
-    const handleSignIn = useCallback((event) => {
-      event.preventDefault()
-      const form = event.target;
-      if(!editMovie){
-        const newMovie = {
-          id: `MO${new Date().valueOf()}VIE`,
-          title: form[0].value,
-          data: form[1].value,
-          url: form[2].value,
-          genre: form[3].value,
-          overview: form[4].value,
-          runtime: form[5].value,
-        };
-        // -----TEST FORM ADDMOVIE SUBMIT-----
-        console.log('TEST FORM ADDMOVIE SUBMIT', newMovie)
-        // --------------------------
-      } else {
-        const editMovie = {
-          id: props.id,
-          title: form[0].value,
-          data: form[1].value,
-          url: form[2].value,
-          genre: form[3].value,
-          overview: form[4].value,
-          runtime: form[5].value,
-        };
-        // -----TEST FORM EDITMOVIE SUBMIT-----
-        console.log('TEST FORM EDITMOVIE SUBMIT', editMovie)
-        // --------------------------
-      }
 
-      handleClose();
-    }, [])
+    const formik = useFormik({
+        initialValues: {
+          ...movie
+        },
+        validationSchema: Yup.object({
+          title: Yup.string()
+            .required('Required'),
+          release_date: Yup.string()
+            .max(10, 'Must be 8 characters')
+            .required('Required'),
+          genres: Yup.string()
+            .max(100, 'Must be 8 characters')
+            .required('Required'),
+          poster_path: Yup.string().url().required('Wrong URL'),
+          overview: Yup.string()
+            .max(500, 'Must be 15 characters or less')
+            .required('Required'),
+          runtime: Yup.number()
+            .positive()
+            .integer()
+            .required('Required'),
+        }),
+        onSubmit: values => {
```

Comment:
`onSubmit` should be a `redux` action or , you can add `editMode` to `initialValues` and get it in the action.
```javascript
const handleSubmit = (values) =>
  handleChangeMovie(values) // <-- redux action
    .then(handleClose)

Reply:
done

Reply:
done